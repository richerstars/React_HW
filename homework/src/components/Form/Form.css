.mainWrapper {
    margin: 200px auto;
    max-width: 600px;
    width: 100%;
    background: #fff;
    border-radius: 5px;
    padding: 46px;
}

.form input {
    outline: none;
    width: 430px;
    height: 60px;
    border: 1px solid #ccc;
    font-size: 17px;
    padding-left: 15px;
    margin: 20px 0 20px 0;
    text-align: center;
}

.form button {
    width: 67px;
    height: 60px;
    border: none;
    background: #f58f28;
    color: #fff;
    font-size: 17px;
    cursor: pointer;
    margin-left: 10px;
}

.form button:disabled,
button[disabled]{
    opacity: 0.3;
    cursor: not-allowed;
}
