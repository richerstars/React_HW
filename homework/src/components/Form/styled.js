import styled from 'styled-components';
import colors from '../constants';

export const StyledInput = styled.input`
  outline: none;
  width: 430px;
  height: 60px;
  border: 1px solid ${colors.inputColor};
  font-size: 25px;
  padding-left: 15px;
  margin: 20px 0;
  text-align: center;
`;

export const StyledButton = styled.button`
  width: 67px;
  height: 60px;
  border: none;
  background: ${colors.buttonColor};
  color: ${colors.semiColor};
  font-size: 22px;
  cursor: pointer;
  margin-left: 10px;
  &:disabled{
    opacity: 0.3;
    cursor: not-allowed;
  }
`;

export  const StyledMainWrapp = styled.div`
  margin: 200px auto;
  max-width: 600px;
  width: 100%;
  background: ${colors.semiColor};
  border-radius: 5px;
  padding: 46px;
`;
export  const StyledTheme = styled.button`
  background: ${colors.buttonColor};
  color: ${colors.semiColor};
  border: none;
  margin-left: 450px;
  height: 30px;
  cursor: pointer;
`;
