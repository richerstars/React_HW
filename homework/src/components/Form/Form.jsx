import React, {Component} from 'react';
import './Form.css';

class Form extends Component {
    constructor(props) {
        super(props);
        this.state = {value: ''};
        this.inputRef = React.createRef();
    }

    componentDidMount() {
        this.inputRef.current.focus();
    }

    handleInputValue = (e) => this.setState({ value: e.target.value });

    handleSubmitValue = () => {
        this.props.addItem({
            title: this.state.value,
            id: Date.now(),
            isChecked: false,
        });
        this.setState({value: ''});
    }

    render() {
        return (
            <div className="form">
                <input ref={this.inputRef} onChange={this.handleInputValue} value={this.state.value} placeholder='Here some text'/>
                <button disabled={!this.state.value.trim()} onClick={this.handleSubmitValue}>ADD</button>
            </div>
        );
    }
}

export default Form;
