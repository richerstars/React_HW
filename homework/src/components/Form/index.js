import {connect} from 'react-redux';
import {addTodo} from "../../store/todos/actions";
import Form from "./Form";
import { showNotification } from "../../store/notification/actions";

const mapDispatchToProps = dispatch => ({
    addTodo: (todo) => dispatch(addTodo(todo)),
    showNotification: (type,message) => dispatch(showNotification(type,message)),
});

export default connect(null, mapDispatchToProps)(Form);
