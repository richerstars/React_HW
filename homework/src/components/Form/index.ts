import {connect} from 'react-redux';
import {Dispatch} from "redux";
import {TTodo} from "../../store/todos/actions";
import {addTodo} from "../../store/todos/actions";
import Form from "./Form";
import { showNotification } from "../../store/notification/actions";
import { TNotify } from '../../store/notification/types';

const mapDispatchToProps = (dispatch: Dispatch) => ({
    addTodo: (todo: TTodo) => dispatch(addTodo(todo)),
    showNotification: (newTask: TNotify) => dispatch(showNotification(newTask)),
});

// @ts-ignore
export default connect(null, mapDispatchToProps)(Form);
