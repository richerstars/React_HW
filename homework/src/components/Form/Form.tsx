import React, { useState, useRef, useEffect } from 'react';
import { StyledInput, StyledButton } from './styled';
import { TTodo } from "../../store/todos/actions";

type TProps = {
    addTodo: (task: TTodo) => void,
    showNotification: (message: { message: string }) => void
};

const Form = ({ addTodo, showNotification }: TProps) => {

    const [value, setValue] = useState('');
    const inputRef = useRef();
    const handleInputValue = (e: React.ChangeEvent<HTMLInputElement>) => {
        setValue(e.target.value);
    };
    useEffect(() => {
        // @ts-ignore
        inputRef.current.focus();
    }, []);

    const handleSubmitValue = () => {
        addTodo({
            value: value,
            id: Date.now(),
            isChecked: false,
        });
        showNotification({ message: `Task checked successfully` });
        setValue('');
    };

    return (
        <>
            <StyledInput ref={ inputRef } onChange={ handleInputValue } value={ value }
                placeholder='Here some text'/>
            <StyledButton disabled={ !value.trim() } onClick={ handleSubmitValue }>ADD</StyledButton>
        </>
    );
};

export default Form;
