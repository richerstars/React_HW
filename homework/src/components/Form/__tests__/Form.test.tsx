import React from 'react';
import {shallow, mount} from 'enzyme';
import Form from "../Form";
import {act} from "react-dom/test-utils";

describe('List', ()=> {
    const props = {
        addTodo: jest.fn(),
        showNotification: jest.fn(),
    };

    const setStateMock = jest.fn();
    beforeEach(() => {
        React.useState = jest.fn().mockReturnValue(['123', setStateMock]);
        Date.now = jest.fn(() => 1487076708000);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render correctly', () => {
        const component = shallow(<Form {...props} />);
        expect(component).toMatchSnapshot();
    });
});
