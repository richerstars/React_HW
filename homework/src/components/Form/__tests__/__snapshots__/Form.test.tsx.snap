// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `ShallowWrapper {}`;

exports[`TaskInput should set in state 1`] = `ShallowWrapper {}`;
